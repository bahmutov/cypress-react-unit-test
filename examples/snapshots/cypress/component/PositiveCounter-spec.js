/// <reference types="cypress" />
import React from 'react'
import { mount } from 'cypress-react-unit-test'
import PositiveCounter from './PositiveCounter'

describe('PositiveCounter', () => {
  it('should render counts', () => {
    mount(<PositiveCounter />)
    cy.get('.increment')
      .click()
      .click()
      .click()
    // make sure the component updates
    cy.contains('Value: 3')
      // previous command yields jQuery element
      // I would like to get its outer HTML which
      // we can do via $el[0].outerHTML shorthand
      .its('0.outerHTML')
      .toMatchSnapshot()

    cy.get('.increment')
      .click()
      .click()

    cy.contains('Value: 5')
      .its('0.outerHTML')
      .toMatchSnapshot()
  })

  it('should not go negative', () => {
    mount(<PositiveCounter />)
    cy.get('.increment').click()
    cy.get('.decrement')
      .click()
      .click()
    cy.contains('Value: 0')
      .its('0.outerHTML')
      .toMatchSnapshot()
  })

  it('snapshots the component state', () => {
    mount(<PositiveCounter />)
    cy.get('.increment')
      .click()
      .click()
      .click()
      .click()
    // The component's code set its reference
    // as a property on the "window" object
    // when running inside Cypress. This allows
    // the test to access it.
    cy.window()
      .its('PositiveCounter.state')
      .toMatchSnapshot()
  })
})
