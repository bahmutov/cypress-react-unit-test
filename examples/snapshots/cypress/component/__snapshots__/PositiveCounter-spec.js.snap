exports[`PositiveCounter > should render counts #0`] = `
<span>Value: 3<button class="decrement">−</button><button class="increment">+</button></span>
`;

exports[`PositiveCounter > should render counts #1`] = `
<span>Value: 5<button class="decrement">−</button><button class="increment">+</button></span>
`;

exports[`PositiveCounter > should not go negative #0`] = `
<span>Value: 0<button class="decrement">−</button><button class="increment">+</button></span>
`;

exports[`PositiveCounter > snapshots the component state #0`] =
{
  "count": 4
};
